tests/js/client/shell/shell-foxx-api-spec.js

```diff
@@ -219,964 +277,645 @@ describe('Foxx service', () => {
-    FoxxManager.install(depPath, mount);
-    const replaceResp = request.put('/_api/foxx/dependencies', {
-      qs: {
-        mount,
-        minimal: false
-      },
-      body: {
+    installFoxx(mount, {type: 'dir', buffer: depPath});
+    const replaceResp = arango.PUT('/_api/foxx/dependencies?mount=' + mount + '&minimal=false', {
         test2: '/test2'
-      },
-      json: true
     });
-    expect(replaceResp.status).to.equal(200);
-    expect(replaceResp.json).to.have.property('test1');
-    expect(replaceResp.json.test1).to.have.property('warning', 'is required');
-    expect(replaceResp.json).to.have.property('test2');
-    expect(replaceResp.json.test2).to.have.property('current', '/test2');
-    const updateResp = request.patch('/_api/foxx/dependencies', {
-      qs: {
-        mount,
-        minimal: false
-      },
-      body: {
+    expect(replaceResp).to.have.property('test1');
+    expect(replaceResp.test1).to.have.property('warning', 'is required');
+    expect(replaceResp).to.have.property('test2');
+    expect(replaceResp.test2).to.have.property('current', '/test2');
+    const updateResp = arango.PATCH('/_api/foxx/dependencies?mount=' + mount + '&minimal=false', {
         test1: '/test1'
-      },
-      json: true
     });
-    expect(updateResp.status).to.equal(200);
-    expect(updateResp.json).to.have.property('test1');
-    expect(updateResp.json.test1).to.have.property('current', '/test1');
-    expect(updateResp.json.test1).to.not.have.property('warning');
-    expect(updateResp.json).to.have.property('test2');
-    expect(updateResp.json.test2).to.have.property('current', '/test2');
-    expect(updateResp.json.test2).to.not.have.property('warning');
-    const resp = request.get('/_api/foxx/dependencies', {qs: {mount, minimal: false}});
-    expect(resp.status).to.equal(200);
-    expect(resp.json).to.have.property('test1');
-    expect(resp.json.test1).to.have.property('current', '/test1');
-    expect(resp.json).to.have.property('test2');
-    expect(resp.json.test2).to.have.property('current', '/test2');
+    expect(updateResp).to.have.property('test1');
+    expect(updateResp.test1).to.have.property('current', '/test1');
+    expect(updateResp.test1).to.not.have.property('warning');
+    expect(updateResp).to.have.property('test2');
+    expect(updateResp.test2).to.have.property('current', '/test2');
+    expect(updateResp.test2).to.not.have.property('warning');
+    const resp = arango.GET('/_api/foxx/dependencies?mount=' + mount + '&minimal=false');
+    expect(resp).to.have.property('test1');
+    expect(resp.test1).to.have.property('current', '/test1');
+    expect(resp).to.have.property('test2');
+    expect(resp.test2).to.have.property('current', '/test2');
   });
 
   it('minimal dependencies should be merged after update', () => {
-    FoxxManager.install(depPath, mount);
-    const replaceResp = request.put('/_api/foxx/dependencies', {
```

Comment:
Does the request module not work when using vst? If so, that should be considered a bug. We're telling users to use the request module for integration tests: https://www.arangodb.com/docs/stable/foxx-guides-testing.html#integration-testing

Reply:
@Simran-B we may want to change this - up to this PR it may however break with binary data.

Reply:
So Foxx+VST is broken in all current releases, but will it stay that way?

Reply:
this PR will fix it. However, using `const request = require("@arangodb/request");` from arangosh which doesn't use fuerte is the primary thing which needs discussion. Encouraging using it in the documentation isn't that clever. I'd change that at least as of 3.6.